examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeDescriptors;
+
+/**
+ * Code snippets used in webdocs.
+ */
+public class Snippets {
+
+  @DefaultCoder(AvroCoder.class)
+  static class Quote {
+    String source;
+    String quote;
+
+    public Quote() {}
+    public Quote(String source, String quote) {
+      this.source = source;
+      this.quote = quote;
+    }
+  }
+
+  /** Using a Read and Write transform to read/write from/to BigQuery. */
+  public static void modelBigQueryIO(Pipeline p) {
+    modelBigQueryIO(p, "", "", "");
+  }
+
+  public static void modelBigQueryIO(Pipeline p,
+      String writeProject, String writeDataset, String writeTable) {
+    {
+      // [START BigQueryTableSpec]
+      String tableSpec = "clouddataflow-readonly:samples.weather_stations";
+      // [END BigQueryTableSpec]
+    }
+
+    {
+      // [START BigQueryTableSpecWithoutProject]
+      String tableSpec = "samples.weather_stations";
+      // [END BigQueryTableSpecWithoutProject]
+    }
+
+    {
+      // [START BigQueryTableSpecObject]
+      TableReference tableSpec = new TableReference()
+          .setProjectId("clouddataflow-readonly")
+          .setDatasetId("samples")
+          .setTableId("weather_stations");
+      // [END BigQueryTableSpecObject]
+    }
+
+    {
+      String tableSpec = "clouddataflow-readonly:samples.weather_stations";
+      // [START BigQueryReadTable]
+      PCollection<Double> maxTemperatures = p
+          .apply(BigQueryIO.readTableRows().from(tableSpec))
+          // Each row is of type TableRow
+          .apply(MapElements.into(TypeDescriptors.doubles()).via(
+              (TableRow row) -> (Double) row.get("max_temperature")));
+      // [END BigQueryReadTable]
+    }
+
+    {
+      // [START BigQueryReadQuery]
+      PCollection<Double> maxTemperatures = p
+          .apply(BigQueryIO.readTableRows().fromQuery(
+              "SELECT max_temperature FROM [clouddataflow-readonly:samples.weather_stations]"))
+          // Each row is of type TableRow
+          .apply(MapElements.into(TypeDescriptors.doubles()).via(
+              (TableRow row) -> (Double) row.get("max_temperature")));
+      // [END BigQueryReadQuery]
+    }
+
+    {
+      // [START BigQueryReadQueryStdSQL]
+      PCollection<Double> maxTemperatures = p
+          .apply(BigQueryIO.readTableRows().fromQuery(
+              "SELECT max_temperature FROM `clouddataflow-readonly.samples.weather_stations`")
+              .usingStandardSql())
+          // Each row is of type TableRow
+          .apply(MapElements.into(TypeDescriptors.doubles()).via(
+              (TableRow row) -> (Double) row.get("max_temperature")));
+      // [END BigQueryReadQueryStdSQL]
+    }
+
+    {
+      // [START BigQuerySchema]
+      String tableSchema = "source:STRING, quote:STRING";
```

Comment:
I suppose this refers to usage of .withJsonSchema(String jsonSchema) - but this string is not in JSON format. What's up with this?

Reply:
Good catch, I copied-pasted the string format from Python. Apparently they are not compatible between Java and Python. Modifying to use actual JSON. Thanks